src/source_util.cpp

```diff
@@ -340,8 +283,11 @@ bool SourceUtil::isPartialMacro(const clang::SourceRange &sourceRange,
 
     // Move up one level closer to the expansion point
 
-    end = sourceManager.getImmediateMacroCallerLoc(end);
```

Comment:
Similarly here, `getImmediateMacroCallerLoc` always returns the beginning of the expansion. It is decomposed here into `getImmediateExpansionRange().getEnd()` instead. This allows the removal of the hack above that skips past the function call arguments.